UX Improvements

metype opened the pull request:
* Display ranks as v-chip elements
* Properly check if a player doesn't exist or a UUID is invalid and tell the user
* Properly display text on error page
* Add a few more dividers in the player profiles to make text not touch edges.

metype committed:
Properly center error page text

metype committed:
Properly check UUID and display ranks as <v-chip>s

Trikolon commented on src/components/PlayerProfile.vue:
```diff
@@ -20,25 +18,23 @@
-                margin-top:10px"
-                v-if="player.connectedTo!=null"
-                >
-                This User Is Online On {{player.connectedTo.name}}.
```
Why are we dropping this completely? I think we should still show if the player is online. Perhaps even with a small visual indicator, like a green circle.

Trikolon commented on src/components/PlayerProfile.vue:
```diff
@@ -20,25 +18,23 @@
                 <br>UUID: {{player.uuid}}</h3>
               <v-divider block style="margin-top:2%;margin-bottom:2%"></v-divider>
-              <h3 class="justify-center"
-                style="text-align:center;">Rank(s): {{playerRanks}}</h3>
```
Please keep the ranks heading. I think "Ranks" would be okay, we don't need to account for the singular case specifically.
Perhaps we can find a nicer way to display headings, not show them inline? Does Vuetify do something similar in the demos we could replicate here?

Trikolon commented on src/components/PlayerProfile.vue:
```diff
@@ -146,17 +137,11 @@ export default {
-        .split(',')
-        .join('] [')
-        .toUpperCase()}]`;
+      if (!this.player?.groups?.length) return ['None'];
```
Should the no-ranks label really be a chip? This might confuse users.

Trikolon commented on src/components/PlayerProfile.vue:
```diff
@@ -127,12 +124,6 @@ export default {
     };
   },
   computed: {
-    playerNotFound() {
```
Nice! Thanks for simplifying this.

Trikolon commented on src/components/PlayerProfile.vue:
```diff
@@ -146,17 +137,11 @@ export default {
-      if (this.player === null || this.player === undefined) return null;
-      if (this.player.infractions === undefined) return null;
-      if (this.player.infractions.bans.length === 0) return null;
+      if (!this.player?.infractions?.bans?.length) return null;
```
You're null checking `this.player` in multiple places. For the player == null case,  would it make sense to use a `v-if="this.player` in the markup so we don't even call these methods in the first place?

Trikolon commented on src/components/PlayerProfile.vue:
```diff
@@ -1,17 +1,15 @@
 <template>
     <v-layout row>
         <v-flex xs12 sm6 offset-sm3>
-            <v-progress-linear indeterminate
```
Can we keep the loading bar?

Trikolon reviewed (changes requested):
I've added some comments. We could also improve UX by structuring the data better. Adding icons, visually separated headings, etc.
For example, the status page is much easier to read that way:
![image](https://user-images.githubusercontent.com/1077860/100549338-b0467480-3272-11eb-9853-7f2bb6130bad.png)

Thanks!

metype committed:
UX Overhaul on Player Profiles & I got carried away

* Player Profiles now user <v-list>s
* Bans now only show if there are any to show
*Bans are no longer a v-data-table, but a grouping of expandable lists
*I copied ideas from the status page on this one because the UX there is really nice in my opinion

metype committed:
Colour icons and make search results a v-item list

metype commented:
I have no issue with it if you'd rather the icons not have colour, I just thought this may make them look a bit nicer.

Trikolon commented on src/components/BanDetail.vue:
```diff
@@ -0,0 +1,56 @@
+
+<script>
+export default {
+  name: 'BanDetail',
```
Can we pass in the invidual fields instead of one big object? So a prop for `server`, one for `banStart` etc.? This way we can also make use of the prop validation.

Trikolon commented on src/components/BanDetail.vue:
```diff
@@ -0,0 +1,56 @@
+  },
+};
+</script>
+<style scoped>
```
nit: You can probably just omit the style section.

Trikolon commented on src/components/PlayerProfile.vue:
```diff
@@ -20,69 +19,122 @@
+                  <v-icon color="light-blue">perm_identity</v-icon>
+                </v-list-item-icon>
+                <v-list-item-content>
+                  <b>Name</b>
```
Can we show the name more prominently? Perhaps as a headline?

Trikolon commented on src/components/PlayerProfile.vue:
```diff
@@ -20,69 +19,122 @@
+                    :ban="ban">
+                  </BanListItem>
+                   </v-expansion-panels>
+                  <!--
```
nit: Please remove code you have commented out.

Trikolon commented on src/components/PlayerSearch.vue:
```diff
@@ -34,17 +34,22 @@
+           v-for="(player, index) in players.result"
+           :key="index"
+           class="row"
+           @click.native="$router.push(`/player/${player.uuid}`)">
```
I think you can use the `to` attribute instead to link to routes. That seems a bit cleaner.

Trikolon reviewed (changes requested):
When I open a player profile page, for example `/player/a98f3718-db9f-4201-b4a8-2dc3e182221d` I get errors in the console. Could you please look into that?

I like the icons! However I think we should not use colors.
The left align of the list doesn't work with our centered layout I think.

Nice use of components!

metype committed:
Remove colours, resolve errors, use primitive props

Trikolon commented on src/components/PlayerSearch.vue:
```diff
@@ -1,13 +1,15 @@
     </v-progress-linear>
-    <div v-if="players.result.length == 0">
+    <div
+     v-if="players.result == undefined || players.result.length==0"
```
nit: `players.result?.length`

metype committed:
Update packages

metype committed:
Change the design on the Player's Profile to center items.

metype committed:
Fix an issue where the text inside the ban details was an h2

metype committed:
/player/[X] where [X] can now be a player name or a UUID, fixed a typo, centered search result list

metype committed:
Prevent negative offset values from being passed into the GraphQl query

metype committed:
Avoid prop mutation in player search and default 'page' to 1, instead of 0

metype committed:
Prevent the page from requesting the skin of 'undefined'

metype committed:
Prevent navigation to current route.

Trikolon reviewed (approved):
Awesome!! Thanks for your work. I have a couple of smaller remarks but I think we can get this merged and maybe address some as a follow-up.
